homeassistant/components/harmony/remote.py

```diff
@@ -193,15 +189,34 @@ def __init__(self, name, host, port, activity, out_path, delay_secs):
         # activity
         await self.new_config()
 
-        async def shutdown(_):
-            """Close connection on shutdown."""
-            _LOGGER.debug("%s: Closing Harmony Hub", self._name)
-            try:
-                await self._client.close()
-            except aioexc.TimeOut:
-                _LOGGER.warning("%s: Disconnect timed-out", self._name)
+    async def shutdown(self):
+        """Close connection on shutdown."""
+        _LOGGER.debug("%s: Closing Harmony Hub", self._name)
+        try:
+            await self._client.close()
+        except aioexc.TimeOut:
+            _LOGGER.warning("%s: Disconnect timed-out", self._name)
 
-        self.hass.bus.async_listen_once(EVENT_HOMEASSISTANT_STOP, shutdown)
+    @property
+    def device_info(self):
+        """Return device info."""
+        model = "Harmony Hub"
+        if "ethernetStatus" in self._client.hub_config.info:
+            model = "Harmony Hub Pro 2400"
+        return {
+            "identifiers": {(DOMAIN, self.unique_id)},
```

Comment:
If the `unique_id` is `None` this id doesn't work. This has to change.

Reply:
unique_id shouldn't ever be none now that the xmpp fix is there.

Reply:
It seems to be able to return `None` here:
https://github.com/home-assistant/core/blob/62f2ee5f60586402e90ac868354550571e96cb2a/homeassistant/components/harmony/util.py#L13

Reply:
I'll remove that.  I should have done that when I fixed the xmpp unique id

Reply:
fixed